src/fparser/common/tests/test_splitline.py

```diff
@@ -165,22 +165,76 @@ def test_splitparen():
     #     print i,l[i],EXPECTED[i],l[i]==EXPECTED[i]
 
 
-def test_splitquote():
-    """Tests splitquote function."""
-    split_list, stopchar = splitquote('abc\\\' def"12\\"3""56"dfad\'a d\'')
-    assert split_list == ["abc\\' def", '"12\\"3"', '"56"', "dfad", "'a d'"]
-    assert stopchar is None
-    result, stopchar = splitquote('abc\\\' def"12\\"3""56"dfad\'a d\'')
-    assert result == ["abc\\' def", '"12\\"3"', '"56"', "dfad", "'a d'"]
-    assert stopchar is None
-
-    split_list, stopchar = splitquote("a'")
-    assert split_list == ["a", "'"]
-    assert stopchar == "'"
-
-    split_list, stopchar = splitquote("a'b")
-    assert split_list == ["a", "'b"]
-    assert stopchar == "'"
+@pytest.mark.parametrize(
+    "input_line, expected_parts, expected_unterm",
+    [
+        # Simple double quoted string
+        ('PRINT *, "Hello"', ["PRINT *, ", '"Hello"'], None),
+        # Simple single quoted string
+        ("PRINT *, 'Hello'", ["PRINT *, ", "'Hello'"], None),
+        # Multiple quoted strings
+        (
+            'PRINT *, "Hello", VAR, "World!"',
+            ["PRINT *, ", '"Hello"', ", VAR, ", '"World!"'],
+            None,
+        ),
+        # Escaped double quotes inside double quoted string
+        (
+            'WRITE(*,*) "He said ""Hello"""',
+            ["WRITE(*,*) ", '"He said ""Hello"""'],
+            None,
+        ),
+        # Escaped single quotes inside single quoted string
+        ("WRITE(*,*) 'It''s fine'", ["WRITE(*,*) ", "'It''s fine'"], None),
+        # Both types in one line
+        ("PRINT *, \"A\", B, 'C'", ["PRINT *, ", '"A"', ", B, ", "'C'"], None),
+        # Mixed with adjacent text
+        ('X="foo""bar"', ["X=", '"foo""bar"'], None),
+        # No quoted strings
+        ("DO 10 I = 1, N", ["DO 10 I = 1, N"], None),
+        # Quoted string at start
+        ('"abc" is a string', ['"abc"', " is a string"], None),
+        # Quoted string at end
+        ('label = "xyz"', ["label = ", '"xyz"'], None),
+        # Embedded commas
+        ('DATA STR /"A,B,C"/', ["DATA STR /", '"A,B,C"', "/"], None),
+        # Fortran character kind (should treat as unquoted)
+        ("character(len=5, kind=1) :: foo", ["character(len=5, kind=1) :: foo"], None),
+        # Unterminated double-quoted string at end
+        ('PRINT *, "unterminated', ["PRINT *, ", '"unterminated'], '"'),
+        # Unterminated single-quoted string at end
+        ("PRINT *, 'unterminated", ["PRINT *, ", "'unterminated"], "'"),
```

Comment:
Can you also add a test for an empty string, e.g. `print *,"test", "", '', "the end"`

I originally thought your new loop would not work, but it does ;)

Reply:
Thanks!